lib/graceful-exit.js

```diff
@@ -148,8 +150,22 @@ exports.middleware = function middleware (app) {
 
   return function checkIfExitingGracefully (req, res, next) {
     if (app.settings.graceful_exit === true) {
-      // sorry keep-alive connections, but we need to part ways
-      req.connection.setTimeout(1);
+      if (options.respondDuringExit) {
+        logger('Server unavailable, incoming request rejected with error: ' + req && req.headers || '?');
+        // Signal the caller that the connection is closing, if before suicideTimeout
+        // 2019-09-10 - HH - Fix request processing after connection closes, see issue 14
+        res.set('Connection', 'close');
+        return next(
+          options.unavailableError() ||
+          defaultOptions.unavailableError(
+              new Error('Server unavailable, no new requests accepted during shutdown')
+            )
+        );
+      } else {
+        logger('Server unavailable, incoming request dropped silently: ' + req && req.headers || '?');
+        res.end(); // silently drop request without response (existing deprecated behavior)
```

Comment:
I'd maybe modify the comment to be something along the lines of:

// silently drop the request, without calling next() so no request code is run, and return no response
// this was the default behavior prior to 0.5.0, with the difference that the request code was called

Reply:
I'll add some context, but generally bug fix information lives in the change log, yes?

Updates to CHANGELOG.md will accompany the version bump.  :-)